h2o-algos/src/main/java/hex/glrm/GLRM.java

```diff
@@ -1131,13 +1186,29 @@ private Chunk chk_xnew(Chunk[] chks, int c) {
 
           // Gradient wrt x_i is matrix product \grad L_{i,j}(x_i * Y_j, A_{i,j}) * Y_j'
           double[] weight = _lossFunc[j].mlgrad(xy, (int) a[j]);
```

Comment:
Moreover, you can add `len` (length of `xy`) as an additional parameter, then `xy` can be reused across iterations as well, provided that you initialize it to length `max(_yt.numLevels)`.